scala/private/rule_impls.bzl

```diff
@@ -229,27 +246,26 @@ StatsfileOutput: {statsfile_output}
       resource_strip_prefix = resource_strip_prefix,
       resource_jars = _join_path(resource_jars),
       dependency_analyzer_mode = dependency_analyzer_mode,
-      statsfile_output = ctx.outputs.statsfile.path)
+      statsfile_output = statsfile.path)
   argfile = ctx.actions.declare_file(
-      "%s_worker_input" % ctx.label.name, sibling = output)
+      "%s_scalac_worker_input" % target_label.name, sibling = output)
 
   ctx.actions.write(
       output = argfile, content = scalac_args + optional_scalac_args)
 
-  outs = [output, ctx.outputs.statsfile]
+  outs = [output, statsfile]
   if buildijar:
     outs.extend([ctx.outputs.ijar])
   ins = (
-      compiler_classpath_jars.to_list() + all_srcjars.to_list() +
-      list(sources) + ctx.files.plugins + dependency_analyzer_plugin_jars +
-      classpath_resources + resources + resource_jars + ctx.files._java_runtime
-      + [ctx.outputs.manifest, ctx.executable._ijar, argfile])
+      compiler_classpath_jars.to_list() + all_srcjars.to_list() + list(sources)
+      + plugins_list + dependency_analyzer_plugin_jars + classpath_resources +
+      resources + resource_jars + [manifest, ctx.executable._ijar, argfile])
```

Comment:
What happened to ctx.files._java_runtime?

Reply:
That should have been removed a long time ago. It was for when we were calling the java compiler ourselves. It is not needed to call ijar which is a c++ program (which we shouldn’t be calling this way anyway).